core/src/main/scala/org/genivi/sota/core/db/Vehicles.scala

```diff
@@ -75,27 +79,29 @@ object Vehicles {
    * objects that reference this vehicle first.
    * @param vehicle The VIN to remove
    */
-  def deleteById(vehicle : Vehicle) : DBIO[Int] = vins.filter( _.vin === vehicle.vin ).delete
+  def deleteById(vehicle : Vehicle) : DBIO[Int] =
+    vehicles.filter(v => v.namespace === vehicle.namespace && v.vin === vehicle.vin).delete
 
   /**
    * Find VINs that match a regular expression
    * @param reg A regular expression
    * @return A list of matching VINs
    */
-  def searchByRegex(reg:String) : Query[VehicleTable, Vehicle, Seq] = vins.filter(vins => regex(vins.vin, reg))
+  def searchByRegex(reg:String): Query[VehicleTable, Vehicle, Seq] =
```

Comment:
we search here across all namespaces; postponed until authN+authZ is implemented